scalafix/rules/src/main/scala/fix/Scalacollectioncompat_newcollections.scala

```diff
@@ -224,11 +224,22 @@ case class Scalacollectioncompat_newcollections(index: SemanticdbIndex)
   // terms dont give us terms https://github.com/scalameta/scalameta/issues/1212
   // if we have a simple identifier, we can look at his definition at query it's type
   // this should be improved in future version of scalameta
-  def isTpe(symbol: Symbol, tree: Tree, ctx: RuleCtx): Boolean =
-    ctx.index.denotation(tree).map(_.names.headOption.exists(_.symbol == symbol)).getOrElse(false)
+  object TypeMatcher {
+    def apply(symbol: Symbol)(implicit index: SemanticdbIndex): TypeMatcher =
+      new TypeMatcher(symbol)(index)
+  }
+
+  final class TypeMatcher(symbol: Symbol)(implicit index: SemanticdbIndex) {
+    def unapply(tree: Tree): Option[Unit] = {
```

Comment:
FYI you can make it return a `Boolean` rather than an `Option[Unit]`. Then at use site just write `case CollectionSet() =>`

Reply:
Yup I know, but I want to have a nested pattern match.

Reply:
:+1: